piptools/writer.py

```diff
@@ -234,6 +236,8 @@ def _format_requirement(
         ireq_hashes = (hashes if hashes is not None else {}).get(ireq)
 
         line = format_requirement(ireq, marker=marker, hashes=ireq_hashes)
+        if self.strip_extras:
+            line = re.sub(r"\[.+?\]", "", line)
```

Comment:
Perhaps, it would make sense to improve `format_requirement(.., strip_extras=True)` and strip `ireq.req.extras` (e.g.,  `ireq.req.extras = set()`) before `str(ireq.req)`. What do you think?

Reply:
I already considered this but looking at the implementation it would require a chain of changes that would make the codebase much uglier. I am not against the idea but I would prefer if this would be done at a later time.

Reply:
Okay, not a reason to block the PR 👍🏻  Nice to know that you've considered it before.